Dashrews/ROX-16326-migration upgrade fix

dashrews78 opened the pull request:
## Description

customer found issue migrating with old cluster health statuses that did not have a UUID as the ID.  Basically they didn't have a UUID.  The copyFrom logic should have logged those and moved on, but the check was in the wrong place.  Moved the check before adding to rows and updated tests to demonstrate this.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Updated migration tests to drive the condition.

dashrews78 committed:
X-Smart-Branch-Parent: master

c-du commented on migrator/migrations/n_10_to_n_11_postgres_cluster_health_statuses/postgres/postgres_plugin.go:
```diff
@@ -139,6 +139,10 @@ func (s *storeImpl) copyFromClusterHealthStatuses(ctx context.Context, tx *postg
 			return marshalErr
 		}
 
+		if pgutils.NilOrUUID(obj.GetId()) == nil {
```
Do we know how does this happen? Why the id is not as expected?

dashrews78 replied:
@c-du Turns out it is because `n_94` used the wrong bucket name

`clusterHealthStatusBucket = []byte("cluster_health_status")`
instead of
`bucket = []byte("clusters_health_status")`

dashrews78 commented:
/retest

dashrews78 committed:
X-Smart-Squash: Squashed 12 commits:

3d352c9512 Updates for migrations so it can not fail copy from when id is not a UUID or nil
905520cdc8 generated tag
6308d1c546 more generated tags
aece9a5fec style
9936e8a747 bunch of style crap
a078f0e92f more style
5a880782ae more style
cee2b3a556 more styling
57bc182d3d more styling
a866681c6f more and more style
a80ae98f1e more style
49d177c419 unused parameters

dashrews78 committed:
more style

dashrews78 commented:
/test gke-nongroovy-e2e-tests

dashrews78 committed:
uncomment writing of object

c-du commented on migrator/migrations/n_52_to_n_53_postgres_simple_access_scopes/postgressimpleaccessscopes/postgres_plugin.go:
```diff
@@ -121,6 +121,11 @@ func (s *storeImpl) copyFromSimpleAccessScopes(ctx context.Context, tx *postgres
 		}
 
+		if pgutils.NilOrUUID(obj.GetId()) == nil {
+			log.Warnf("Id is not a valid uuid -- %v", obj)
```
We still need to use utils.Should.

c-du commented on migrator/migrations/n_52_to_n_53_postgres_simple_access_scopes/postgrespermissionsets/postgres_plugin.go:
```diff
@@ -121,6 +121,11 @@ func (s *storeImpl) copyFromPermissionSets(ctx context.Context, tx *postgres.Tx,
 		}
 
+		if pgutils.NilOrUUID(obj.GetId()) == nil {
+			log.Warnf("Id is not a valid uuid -- %v", obj)
```
utils.Should

connorgorman commented on migrator/migrations/n_42_to_n_43_postgres_process_indicators/postgres/postgres_plugin.go:
```diff
@@ -153,6 +151,11 @@ func (s *storeImpl) copyFromProcessIndicators(ctx context.Context, tx *postgres.
 		}
 
+		if pgutils.NilOrUUID(obj.GetId()) == nil {
+			log.Warnf("Id is not a valid uuid -- %v", obj)
```
Don't print a process indicator as it can potentially have secrets in the command line. Perhaps the same applies for secret below even though we don't take the data fields

connorgorman commented on migrator/migrations/n_23_to_n_24_postgres_image_integrations/postgres/postgres_plugin.go:
```diff
@@ -127,6 +125,11 @@ func (s *storeImpl) copyFromImageIntegrations(ctx context.Context, tx *postgres.
 		}
 
+		if pgutils.NilOrUUID(obj.GetId()) == nil {
+			log.Warnf("Id is not a valid uuid -- %v", obj)
```
Definitely do not log this

connorgorman reviewed (changes requested):
Evaluate the log lines for sensitive data

dashrews78 committed:
clean up logging

dashrews78 commented on migrator/migrations/n_03_to_n_04_postgres_deployments/postgres/postgres_plugin.go:
```diff
@@ -148,8 +146,8 @@ func insertIntoDeploymentsContainers(ctx context.Context, batch *pgx.Batch, obj
 	}
-	if pgutils.NilOrUUID(deployments_Id) == nil {
-		utils.Should(errors.Errorf("deployments_Id is not a valid uuid -- %v", obj))
+	if pgutils.NilOrUUID(deploymentID) == nil {
```
since I had to go back in for logging the object, I took care of this while I was there.

dashrews78 committed:
adjust value logged
